### ISSUE TITLE
Can't get latest 'search' system to do anything.

### ISSUE THREAD

**Issue by alan666notb** [cgeo/cgeo#11360]
### Describe your problem!

I can't get any results out of the 'search' system on the latest release.

If I go to the search page and start typing "church" into the keyword filter, a list of caches pop-up showing those with "ch", "chu" "chur" etc. in them, but when I hit "Search by keyword" I get "No caches here"

The filter line at the top says "-|Cache name:contains:church"

I get the same result (i.e. no results) searching 'Hidden by'.

My partner, who is running the same version, get the expected list of caches. I can't see any differences between our setups.

### How to reproduce?

Goto 'Search' page,
type text into 'Keyword' field,
Hit 'Search by keyword'.

### Reproducible

Yes

### c:geo Version

2021.07.28-RC

### System information

```shell
## System information

c:geo version: 2021.07.28-RC

Device:
-------
- Device type: M2101K6G (sweet_eea, Redmi)
- Available processors: 8
- Android version: 11
- Android build: RKQ1.200826.002 test-keys
- Screen resolution: 1080x2177px (392x791dp)
- Sailfish OS detected: false
- Google Play services: enabled - 21.24.18 (150400-383468479)
- HW acceleration: enabled (default state)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Filters: 
    LIVE: None ([:inconclusive=false:advanced=false]AND(type;difficulty_terrain:difficulty=-:difficulty=-:terrain=-:terrain=-;status))
    OFFLINE: None ([:inconclusive=false:advanced=false])
- Hide waypoints: -
- Set language: en_GB (system default)
- System date format: dd/MM/y
- Debug mode active: no
- Live map mode: false
- OSM multi-threading: true / threads: 4
- Last backup: 10 Jul, 08:38
- Routing mode: Off
- Settings: v8, Count:198
- Map: Great_Britain_OAM_V4_May2021 (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:primary:OpenStreetMaps/Great_Britain_OAM_V4_May2021.map
  - Atts: Maps: (c)Christian Kernbeis, Themes: (c)Tobias Kuehn, www.openandromaps.org 2012-2021
  - Theme: ElevateTheme.zip:Elevate.xml

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Login OK) / PREMIUM
- Geocaching.com date format: dd MMM yy
- Routing: external / BRouter installed: false
- Installed c:geo plugins:  none

Permissions & paths:
-------
- Fine location permission: granted
- Write external storage permission: granted
- System internal c:geo dir: /data/user/0/cgeo.geocaching (56.4 GB free) v2 internal isDir(7 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (56.4 GB free) v2 internal isDir(7 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (56.4 GB free) v2 external non-removable isDir(1138 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (56.4 GB free) v2 internal isDir(0 entries)
- Public Folders: #11
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=6, dirs:>=12, totalFileSize:>=109.5 MB, free space: 56.4 GB, files on device: 13511677)
  - OFFLINE_MAPS: /OpenStreetMaps (User-Defined)[/OpenStreetMaps[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3AOpenStreetMaps::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AOpenStreetMaps/document/primary%3AOpenStreetMaps, Av:true, files:6, dirs:1, totalFileSize:3.2 GB, free space: 56.4 GB, files on device: 13511677)
  - OFFLINE_MAP_THEMES: /OpenStreetMaps (User-Defined)[/OpenStreetMaps[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3AOpenStreetMaps::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AOpenStreetMaps/document/primary%3AOpenStreetMaps, Av:true, files:6, dirs:1, totalFileSize:3.2 GB, free space: 56.4 GB, files on device: 13511677)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 56.4 GB, files on device: 13511677)
  - GPX: /My Documents/SyncedWithGoogle/GPX (User-Defined)[/My Documents/SyncedWithGoogle/GPX[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3AMy%20Documents%2FSyncedWithGoogle%2FGPX::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AMy%20Documents%2FSyncedWithGoogle%2FGPX/document/primary%3AMy%20Documents%2FSyncedWithGoogle%2FGPX, Av:true, files:16, dirs:2, totalFileSize:23.0 MB, free space: 56.4 GB, files on device: 13511677)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:>=10, dirs:>=5, totalFileSize:>=177.1 MB, free space: 56.4 GB, files on device: 13511677)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 56.4 GB, files on device: 13511677)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 56.4 GB, files on device: 13511677)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:7, dirs:1, totalFileSize:81.2 KB, free space: 56.4 GB, files on device: 13511677)
  - ROUTING_TILES: /cgeo/routing/segments4 (Default)[/cgeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 56.4 GB, files on device: 13511677)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 56.4 GB, files on device: -1)
- Map render theme path: ElevateTheme.zip:Elevate.xml
- PersistedDocumentUris: #1
- TRACK: null
- Persisted Uri Permissions: #3
  - content://com.android.externalstorage.documents/tree/primary%3AMy%20Documents%2FSyncedWithGoogle%2FGPX (26 May, 15:31):RW
  - content://com.android.externalstorage.documents/tree/primary%3AOpenStreetMaps (7 May, 19:56):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (7 May, 19:45):RW
- Database: /storage/emulated/0/Android/data/cgeo.geocaching/files/databases/data (v96, Size:33.4 MB) on user storage

--- End of system information ---
```

**Comment by eddiemuc** [comment]
We had the same error before, it vanished when the user opened the live filter, switched to "advanced" mode, deleted the terrain/difficulty filter and saved it.

But before you do that: would it be possible for you to create and send an extended logfile first? (Via settings -> System -> create logfile)

**Comment by eddiemuc** [comment]
Might be same root cause as for #11331

**Comment by alan666notb** [comment]
Hi @eddiemuc - I was just reading 113311 when I saw your reference appear!

I've created an extended log file, 
[logcat-2021-07-29-17-30-05-2.txt](https://github.com/cgeo/cgeo/files/6902094/logcat-2021-07-29-17-30-05-2.txt)

**Comment by alan666notb** [comment]
@eddiemuc - some more info, 

I was playing with the D/T filter on the live map before I found this bug, I see that's mentioned in #11331 so that could be a connection.
I haven't cleared the D/T filter yet to 'test' it, I'll leave the bug 'in place' in case I can help with any extra info.

**Comment by eddiemuc** [comment]
I am pretty confident the error lies in `DifficultyAndTerrainFilter` class in the way it creates it's "subfolders" `DifficultyFilter` and `Terrain filter` in lines 11 and 12. Instead of instantiating them directly, they should use the `create` method of enum  `CacheFilterType` (otherwise the type field of the filter class is not initialized).

I have no access this evening to a computer. If this is urgent, can someone please step in?
Maybe @murggel?

**Comment by murggel** [comment]
> I am pretty confident the error lies in `DifficultyAndTerrainFilter` class in the way it creates it's "subfolders" `DifficultyFilter` and `Terrain filter` in lines 11 and 12. Instead of instantiating them directly, they should use the `create` method of enum  `CacheFilterType` (otherwise the type field of the filter class is not initialized).
> 
> I have no access this evening to a computer. If this is urgent, can someone please step in?
> Maybe @murggel?

Sure, will have a look. I tried to use the create-method, but don't know, what was reason I didn't use it in the end.

**Comment by Lineflyer** [comment]
Similar reports for current beta on support mail.
Most cases the searches only show adventure lab caches (perhaps as they do not have a D/T).

**Comment by murggel** [comment]
assumption of @eddiemuc is right, PR is on its way.

**Comment by fm-sys** [comment]
Fixed with  #11361. Commits are merged back to master, but auto close failed as the "fix #xxxx" statement was only part of the PR description, but not contained in the commit message...
